File: LATIHAN2DPBO2023-main/PHP/CR.php
<?php 
// Saya Raisyad Jullfikar NIM 2106238 mengerjakan Latihan 2 
// dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
// keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

  // Import File Mahasiswa
  require('Mahasiswa.php');
  class CR {
    // Deklar Private arr
    private $data = array();
    
    // Constructor
    public function __construct() {}

    // Method untuk menginputkan data
    public function create ($idx, $NIK, $name, $jenis_kelamin, $asal_universitas, $email_edu, $NIM, $fakultas, $prodi, $pasfoto) {
      $this->data[$idx] = new Mahasiswa(); // deklar arr untuk menampung objek mahasiswa
      
      // Assign value to array var
      $this->data[$idx]->setNik($NIK);
      $this->data[$idx]->setName($name);
      $this->data[$idx]->setJKelamin($jenis_kelamin);
      $this->data[$idx]->setAUniv($asal_universitas);
      $this->data[$idx]->setEmail($email_edu);
      $this->data[$idx]->setNim($NIM);
      $this->data[$idx]->setFakultas($fakultas);
      $this->data[$idx]->setProdi($prodi);
      $this->data[$idx]->setFoto($pasfoto);
    }
    
    // Method untuk read/show/view the data
    public function read () {
      echo  "<h4 style='text-align: center;'>Data Sesudah CR</h4>" .
              "<table border='1' cellspacing='0' cellpadding='10'>" .
              "<tr>" .
              "<th>No.</th>" .
              "<th>NIK</th>" .
              "<th>Name</th>" .
              "<th>Jenis Kelamin</th>" .
              "<th>Asal Universitas</th>" .
              "<th>Email</th>" .
              "<th>NIM</th>" .
              "<th>Fakultas</th>" .
              "<th>Prodi</th>" .
              "<th>Pas Foto</th>" .
              "</tr>";
        for ($n = 0; $n < count($this->data); $n++) {
          echo  "<tr>" .
                "<td>" . ($n+1) . "</td>" .
                "<td>" . $this->data[$n]->getNIK() . "</td>" .
                "<td>" . $this->data[$n]->getName() . "</td>" .
                "<td>" . $this->data[$n]->getJKelamin() . "</td>" .
                "<td>" . $this->data[$n]->getAUniv() . "</td>" .
                "<td>" . $this->data[$n]->getEmail() . "</td>" .
                "<td>" . $this->data[$n]->getNim() . "</td>" .
                "<td>" . $this->data[$n]->getFakultas() . "</td>" .
                "<td>" . $this->data[$n]->getProdi() . "</td>" .
                "<td>" . "<img src='" . $this->data[$n]->getFoto() . "' style='width: 150px;'>" . "</td>" .
                "</tr>";
        }
        echo "</table>";
    }

    // Destructor
    public function __destruct() {}
  }
?>

File: LATIHAN2DPBO2023-main/Java/CR.java

// Saya Raisyad Jullfikar NIM 2106238 mengerjakan Latihan 2 
// dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
// keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

// Import Library
import java.util.*;

// Class Cr yang berisi data list dari class mahasiswa yang akan
// digunakan untuk menginputkan dan menampilkan data list tsb, seperti
// Create, Read/Show
public class CR {
  // Private attr string
  private String NIK,
      name,
      jenis_kelamin,
      asal_universitas,
      email_edu,
      NIM,
      fakultas,
      prodi;

  // Construct kosong
  public CR() {
  }

  // Method Create Data
  public void Create(ArrayList<Mahasiswa> list) {
    Scanner sc = new Scanner(System.in).useLocale(Locale.US);
    Mahasiswa dataIn = new Mahasiswa();
    System.out.println("Input Data NIK:");
    NIK = sc.next();

    System.out.println("Input Data Nama:");
    name = sc.next();

    System.out.println("Input Data Jenis Kelamin/Gender:");
    jenis_kelamin = sc.next();

    System.out.println("Input Data Asal Universitas:");
    asal_universitas = sc.next();

    System.out.println("Input Data Email:");
    email_edu = sc.next();

    System.out.println("Input Data NIM:");
    NIM = sc.next();

    System.out.println("Input Data Fakultas:");
    fakultas = sc.next();

    System.out.println("Input Data Prodi:");
    prodi = sc.next();

    dataIn.setNik(NIK);
    dataIn.setName(name);
    dataIn.setJkelamin(jenis_kelamin);
    dataIn.setAUniv(asal_universitas);
    dataIn.setEmail(email_edu);
    dataIn.setNim(NIM);
    dataIn.setFakultas(fakultas);
    dataIn.setProdi(prodi);

    list.add(dataIn);
  }

  // Method Read Data
  public void Read(ArrayList<Mahasiswa> list) {
    System.out.println("\nList Data yang telah diinputkan: \n");
    for (int i = 0; i < list.size(); i++) {
      System.out.println(Integer.toString(i + 1) + ". "
          + list.get(i).getNik() + " | "
          + list.get(i).getName() + " | "
          + list.get(i).getJKelamin() + " | "
          + list.get(i).getAUniv() + " | "
          + list.get(i).getEmail() + " | "
          + list.get(i).getNim() + " | "
          + list.get(i).getFakultas() + " | "
          + list.get(i).getProdi() + " | ");
    }
    System.out.print("\n");
  }
}
